src/test/java/api/support/APITests.java

```diff
@@ -251,105 +257,47 @@
   protected final FeeFineAccountFixture feeFineAccountFixture = new FeeFineAccountFixture();
 
   protected APITests() {
-    this(true);
+    this(true, false);
   }
 
-  protected APITests(boolean initialiseCirculationRules) {
+  protected APITests(boolean initialiseCirculationRules, boolean enableLoanHistory) {
     this.initialiseCirculationRules = initialiseCirculationRules;
+    setLoanHistoryEnabled(enableLoanHistory);
   }
 
   @BeforeClass
   public static void beforeAll() throws InterruptedException, ExecutionException,
     TimeoutException {
 
-    deployVerticles();
+    if (okapiAlreadyDeployed) {
+      return;
+    }
 
-    //Delete everything first just in case
-    deleteAllRecords();
```

Comment:
Why are the records no longer deleted at the beginning?

Reply:
It was replaced by removing all records before each method.

I've got series of hanged builds:
![image](https://user-images.githubusercontent.com/9094650/83140224-667ba680-a0f6-11ea-84a4-ee1c038cf4d7.png)

Seems the reasons is that:
* There are a lot of network interaction with vert.x, so it is just stuck.
* It also stopped to work, when we re-deploy verticle before each test class; 

Changes that I've applied:
* Deploy verticle only once;
* Introduce a shared storage, so it can be cleaned up without network interaction.

Reply:
Do we know why this has suddenly changed? What about this pull request triggered this to happen?

> Deploy verticle only once

Was it being deployed for each test class?

> Introduce a shared storage, so it can be cleaned up without network interaction

This is a significant trade off, it now means the tests are actively coupled to the fake implementations, making it harder and less likely that these tests can be used with real modules again. Ah well :-/

Reply:
> Do we know why this has suddenly changed? What about this pull request triggered this to happen?

This PR introduced loan history for each loan (means there was a pre-processor that POSTed a request every time a loan created/updated). That was the first failed build. But even when I disabled this function the build is not become stable.

> This is a significant trade off, it now means the tests are actively coupled to the fake implementations, making it harder and less likely that these tests can be used with real modules again. Ah well :-/

Hmm... we can introduce a fake endpoint (e.g. `DELETE /_/tenant`)  that will remove all the data instead of querying storage implementation.

Reply:
I've introduced the /_/tenant to remove all data.

Reply:
Unfortunately, I don't think that will help the challenge, as that endpoint does not exist in the real system, and so the mismatch remains.

Apologies if I explained the challenge poorly.

Reply:
I think I probably need to accept that my experiment with this style of testing is over and to move on